src/main/scala/nl.knaw.dans.easy.bagstore/component/BagStoreComponent.scala

```diff
@@ -173,6 +173,19 @@ trait BagStoreComponent {
       }
     }
 
+    def getSize(itemId: ItemId): Try[Long] = {
+      trace(itemId)
+      val bagId = BagId(itemId.uuid)
+
+      fileSystem.checkBagExists(bagId).flatMap { _ =>
```

Comment:
Include this as the first line of the for-comprehension:

```scala
for {
    _ <- fileSystem.checkBagExists(bagId)
    bagDir <- fileSystem.toLocation(bagId)
    ...
```

do so in the above function `copyToStream` as well. You probably copy-pasted a lot from there to here.